pandas/core/indexes/base.py

```diff
@@ -1311,6 +1312,28 @@ def _get_names(self):
         return FrozenList((self.name, ))
 
     def _set_names(self, values, level=None):
+        """
```

Comment:
can you also add a mention on ``set_names`` itself that the names must be hashable (and examples if you want)